src/loader.js

```diff
@@ -362,7 +362,7 @@ function setupModuleLoader(window) {
           component: function(name, options) {
             function factory($injector) {
               function makeInjectable(fn) {
-                if (angular.isFunction(fn)) {
+                if (isFunction(fn) || isArray(fn)) {
```

Comment:
I am pretty sure we cannot rely on being in the closure of angular here.
Although the unit tests will pass, I don't think that `isFunction` and `isArray` will be available in practice.